### PR TITLE
build(tsc): Replace webpack with tsc

### PR DESCRIPTION
**Pull request by ElijahKotyluk**
## Task List:
 - [x] Remove webpack(Uninstall/Delete config file)
 - [x] Update build script in all packages
 - [x] Change TS config module opt to `CommonJS`

### CODE REVIEW

**Review by dsseng — CHANGES_REQUESTED** [review]
Please  update snapshots

File: packages/core/src/test.ts [hunk 1]
```diff
@@ -24,7 +23,7 @@ export default class Test extends Runnable {
     try {
       this.fn();
     } catch (error) {
-      if (error instanceof ExpectError) {
```
Comment (dsseng) [inline packages/core/src/test.ts:27]: This should be left in place. (if some assert failed, test is marked as failed, if just an error happened: errored). but maybe we should remove errored status in future.

File: packages/core/tsconfig.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "compilerOptions": {
     "outDir": "./dist/",
     "rootDir": "./src",
-    "module": "es2015",
+    "module": "CommonJS",
```
Comment (dsseng) [inline packages/core/tsconfig.json]:
```suggestion
    "module": "commonjs",
```

File: packages/matchers/tsconfig.json [hunk 3]
```diff
@@ -2,7 +2,7 @@
   "compilerOptions": {
     "outDir": "./dist/",
     "rootDir": "./src",
-    "module": "es2015",
+    "module": "CommonJS",
```
Comment (dsseng) [inline packages/matchers/tsconfig.json]:
```suggestion
    "module": "commonjs",
```

File: packages/mock/tsconfig.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
   "compilerOptions": {
     "outDir": "./dist/",
     "rootDir": "./src",
-    "module": "es2015",
+    "module": "CommonJS",
```
Comment (dsseng) [inline packages/mock/tsconfig.json]:
```suggestion
    "module": "commonjs",
```

**Review by dsseng — CHANGES_REQUESTED** [review]
please update snapshots

**Comment by ElijahKotyluk** [comment]
Snapshots are already updated.

**Comment by dsseng** [comment]
They should be updated once more since you fixed `ExpectError` issue.

**Pull request merged**